src/Shared/LayoutRenderers/AspNetUserClaimLayoutRenderer.cs

```diff
@@ -51,29 +58,61 @@ protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
         {
             try
             {
-                var claimsPrincipel = HttpContextAccessor.HttpContext.User;
-                if (claimsPrincipel == null)
+                var claimsPrincipal = HttpContextAccessor.HttpContext.User;
+                if (claimsPrincipal == null)
                 {
                     InternalLogger.Debug("aspnet-user-claim - HttpContext User is null");
                     return;
                 }
 
-                var claimsIdentity = claimsPrincipel.Identity as ClaimsIdentity;    // Prioritize primary identity
-                var claim = claimsIdentity?.FindFirst(ClaimType)
-#if ASP_NET_CORE
-                    ?? claimsPrincipel.FindFirst(ClaimType)
+                if (All)
+                {
+                    var claimKeyValuePairs =
```

Comment:
Could this be extracted into a method called GetAllClaims(), and instead of allocating a list then return Linq-IEnumerable or yield return.
```c#
var claims = GetAllClaims(claimsPrincipal);
SerializePairs(claimKeyValuePairs, builder, logEvent);
```

Reply:
Completed